src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        private Node<T> next;
+        private Node<T> prev;
+        private T data;
+
+        Node(T data) {
+            this.data = data;
+        }
+    }
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        addLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if (index == size) {
+            addLast(value);
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> newNode = new Node<>(value);
```

Comment:
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/d05f1084-a5cd-4137-a876-36bc9096b774)

Reply:
not fixed